resolver/src/p2p/fetcher.rs

```diff
@@ -52,7 +52,8 @@ pub struct Fetcher<
 
     /// Manages pending requests. If fetches fail to make a request to a peer, they are instead
     /// added to this map and are retried after the deadline.
-    pending: PrioritySet<Key, SystemTime>,
+    pending: PrioritySet<Key, (SystemTime, bool)>,
+    waiter: Option<SystemTime>,
```

Comment:
Ideally expand the comment a bit to explain what the bool is and what the waiter does

Since the boolean is unnamed (is not called `is_retry` or `is_new`) then you might want to use an enum with two values instead for readability